--- a/meson.build
+++ b/meson.build
@@ -40,7 +40,7 @@
 else
   dep_pthread_stubs = []
 endif
-dep_threads = dependency('threads')
+dep_threads = dependency('threads', static : true)
 
 cc = meson.get_compiler('c')
 
@@ -237,7 +237,7 @@
 libdrm_c_args = warn_c_args + ['-fvisibility=hidden']
 
 
-dep_pciaccess = dependency('pciaccess', version : '>= 0.10', required : with_intel)
+dep_pciaccess = dependency('pciaccess', version : '>= 0.10', required : with_intel, static : true)
 dep_cunit = dependency('cunit', version : '>= 2.1', required : false)
 _cairo_tests = get_option('cairo-tests')
 if _cairo_tests != 'false'
@@ -303,6 +303,21 @@
 
 inc_root = include_directories('.')
 inc_drm = include_directories('include/drm')
+
+libdrm = static_library(
+  'drm',
+  [files(
+    'xf86drm.c', 'xf86drmHash.c', 'xf86drmRandom.c', 'xf86drmSL.c',
+    'xf86drmMode.c'
+    ),
+    config_file,
+    ],
+  c_args : libdrm_c_args,
+  dependencies : [dep_valgrind, dep_rt, dep_m],
+  include_directories : inc_drm,
+  version : '2.4.0',
+  install : true,
+)
 
 libdrm = shared_library(
   'drm',
