diff --git a/makefile b/makefile
index 0240e5e..8a0dd64 100644
--- a/makefile
+++ b/makefile
@@ -38,7 +38,7 @@ CC = $(HOSTCC)  # change this is cross-compiling.
 # By fiat, to make our lives easier, yacc is now defined to be bison.
 # If you want something else, you're on your own.
 # YACC = yacc -d -b awkgram
-YACC = bison -d
+YACC = byacc
 
 OFILES = b.o main.o parse.o proctab.o tran.o lib.o run.o lex.o
 
@@ -57,7 +57,7 @@ a.out:	awkgram.tab.o $(OFILES)
 $(OFILES):	awk.h awkgram.tab.h proto.h
 
 awkgram.tab.c awkgram.tab.h:	awk.h proto.h awkgram.y
-	$(YACC) $(YFLAGS) awkgram.y
+	$(YACC) $(YFLAGS) -d -o awkgram.tab.c awkgram.y
 
 proctab.c:	maketab
 	./maketab awkgram.tab.h >proctab.c
