--- a/pl/meson.build
+++ b/pl/meson.build
@@ -14,7 +14,6 @@
 )
 
 wayfire = dependency('wayfire')
-giomm = dependency('giomm-2.4')
 wlroots = dependency('wlroots')
 wfconfig = dependency('wf-config')
 cairo = dependency('cairo')

--- a/pl/src/meson.build
+++ b/pl/src/meson.build
@@ -2,10 +2,6 @@
     dependencies: [wayfire, wlroots, wfconfig, cairo],
     install: true, install_dir: join_paths(get_option('libdir'), 'wayfire'))
 
-autorotate = shared_module('autorotate-iio', 'autorotate-iio.cpp',
-    dependencies: [wayfire, wlroots, wfconfig, giomm],
-    install: true, install_dir: join_paths(get_option('libdir'), 'wayfire'))
-
 background_view = shared_module('background-view', 'background-view.cpp',
     dependencies: [wayfire, wlroots, wfconfig],
     install: true, install_dir: join_paths(get_option('libdir'), 'wayfire'))
@@ -26,10 +22,6 @@
     dependencies: [wayfire, wlroots, wfconfig],
     install: true, install_dir: join_paths(get_option('libdir'), 'wayfire'))
 
-glib_main_loop = shared_module('glib-main-loop', 'glib-main-loop.cpp',
-    dependencies: [wayfire, wlroots, wfconfig, giomm],
-    install: true, install_dir: join_paths(get_option('libdir'), 'wayfire'))
-
 joinviews = shared_module('join-views', 'join-views.cpp',
     dependencies: [wayfire, wlroots, wfconfig],
     install: true, install_dir: join_paths(get_option('libdir'), 'wayfire'))
