--- a/src/core/core.cpp
+++ b/src/core/core.cpp
@@ -294,12 +294,6 @@
     protocols.presentation = wlr_presentation_create(display, backend);
     protocols.viewporter   = wlr_viewporter_create(display);
 
-    protocols.foreign_registry = wlr_xdg_foreign_registry_create(display);
-    protocols.foreign_v1 = wlr_xdg_foreign_v1_create(display,
-        protocols.foreign_registry);
-    protocols.foreign_v2 = wlr_xdg_foreign_v2_create(display,
-        protocols.foreign_registry);
-
     wf_shell  = wayfire_shell_create(display);
     gtk_shell = wf_gtk_shell_create(display);
 
