--- a/meson.build
+++ b/meson.build
@@ -203,15 +203,15 @@
 pango_deps += mathlib_dep
 
 # gobject
-glib_dep = dependency('glib-2.0', version: glib_req_version,
+glib_dep = dependency('glib-2.0', static: true, version: glib_req_version,
                       fallback: ['glib', 'libglib_dep'])
-gobject_dep = dependency('gobject-2.0', version: glib_req_version,
+gobject_dep = dependency('gobject-2.0', static: true, version: glib_req_version,
                          fallback: ['glib', 'libgobject_dep'])
-gio_dep = dependency('gio-2.0', version: glib_req_version,
+gio_dep = dependency('gio-2.0', static: true, version: glib_req_version,
                          fallback: ['glib', 'libgio_dep'])
 pango_deps += [glib_dep, gobject_dep, gio_dep]
 
-fribidi_dep = dependency('fribidi', version: fribidi_req_version,
+fribidi_dep = dependency('fribidi', static: true, version: fribidi_req_version,
                          fallback: ['fribidi', 'libfribidi_dep'],
                          default_options: ['docs=false'])
 pango_deps += fribidi_dep
@@ -239,7 +239,7 @@
 harfbuzz_lib=''
 fontconfig_lib=''
 
-harfbuzz_dep = dependency('harfbuzz', version: harfbuzz_req_version, required: false)
+harfbuzz_dep = dependency('harfbuzz', static: true, version: harfbuzz_req_version, required: false)
 if harfbuzz_dep.found()
   harfbuzz_pc = 'harfbuzz'
 else
@@ -259,7 +259,7 @@
 # Remove when Meson acquires ability to declare deps declaratively, or
 # when finding dependencies via CMake files is fixed.
 if not harfbuzz_dep.found()
-  harfbuzz_dep = dependency('harfbuzz', version: harfbuzz_req_version,
+  harfbuzz_dep = dependency('harfbuzz', static: true, version: harfbuzz_req_version,
                             fallback: ['harfbuzz', 'libharfbuzz_dep'])
 endif
 
@@ -278,7 +278,7 @@
   fontconfig_option = false
 endif
 
-fontconfig_dep = dependency('fontconfig', version: fontconfig_req_version, required: fontconfig_option)
+fontconfig_dep = dependency('fontconfig', static: true, version: fontconfig_req_version, required: fontconfig_option)
 if fontconfig_dep.found()
   fontconfig_pc = 'fontconfig'
 else
@@ -296,7 +296,7 @@
 # user option on platforms that always require fontconfig.
 fontconfig_option = fontconfig_required ? true : get_option('fontconfig')
 if not fontconfig_dep.found()
-  fontconfig_dep = dependency('fontconfig', version: fontconfig_req_version,
+  fontconfig_dep = dependency('fontconfig', static: true, version: fontconfig_req_version,
                               fallback: ['fontconfig', 'fontconfig_dep'],
                               required: fontconfig_option)
 endif
@@ -330,7 +330,7 @@
 endif
 
 # The first version of freetype with a pkg-config file is 2.1.5
-freetype_dep = dependency('freetype2', required: freetype_option)
+freetype_dep = dependency('freetype2', static: true, required: freetype_option)
 
 if freetype_dep.found()
   freetype2_pc = 'freetype2'
@@ -350,7 +350,7 @@
 # Do the fallback now if freetype2 has not been found on the system.
 freetype_option = freetype_required ? true : get_option('freetype')
 if not freetype_dep.found()
-  freetype_dep = dependency('freetype2', required: freetype_option,
+  freetype_dep = dependency('freetype2', static: true, required: freetype_option,
                             fallback: ['freetype2', 'freetype_dep'])
 endif
 
@@ -391,7 +391,7 @@
 endif
 
 cairo_found_type = ''
-cairo_dep = dependency('cairo', version: cairo_req_version, required: cairo_option)
+cairo_dep = dependency('cairo', static: true, version: cairo_req_version, required: cairo_option)
 
 if cairo_dep.found()
   cairo_found_type = cairo_dep.type_name()
@@ -405,7 +405,7 @@
 # Remove once Meson gains capability to declare dependencies
 # in a declarative way
 if not cairo_dep.found()
-  cairo_dep = dependency('cairo', version: cairo_req_version,
+  cairo_dep = dependency('cairo', static: true, version: cairo_req_version,
                          fallback: ['cairo', 'libcairo_dep'], required: get_option('cairo'),
                          default_options: ['freetype=enabled', 'fontconfig=enabled'])
   cairo_found_type = cairo_dep.type_name()
@@ -434,9 +434,9 @@
 
   foreach b: cairo_font_backends
     if cairo_found_type == 'pkgconfig'
-      dep = dependency(b[0], version: b[1], required: false)
+      dep = dependency(b[0], static: true, version: b[1], required: false)
     else
-      dep = dependency('', required: false)
+      dep = dependency('', static: true, required: false)
       foreach f: cairo_features
         if f['name'] == b[0]
           dep = cairo_dep
@@ -475,9 +475,9 @@
 
   foreach b: cairo_surface_backends
     if cairo_found_type == 'pkgconfig'
-      dep = dependency(b[0], version: b[1], required: false)
+      dep = dependency(b[0], static: true, version: b[1], required: false)
     else
-      dep = dependency('', required: false)
+      dep = dependency('', static: true, required: false)
       foreach f: cairo_features
         if f['name'] == b[0]
           dep = cairo_dep
