--- a/fc-cache/meson.build
+++ b/fc-cache/meson.build
@@ -6,8 +6,3 @@
 )
 
 tools_man_pages += ['fc-cache']
-
-# Do not try to execute target's fc-cache on host when cross compiling
-if not meson.is_cross_build()
-  meson.add_install_script('../install-cache.py', fccache.full_path())
-endif
--- a/conf.d/meson.build
+++ b/conf.d/meson.build
@@ -61,12 +61,6 @@
 
 install_data(conf_files, install_dir: join_paths(get_option('datadir'), 'fontconfig/conf.avail'))
 
-meson.add_install_script('link_confs.py',
-  join_paths(get_option('prefix'), get_option('datadir'), 'fontconfig/conf.avail'),
-  join_paths(get_option('sysconfdir'), 'fonts', 'conf.d'),
-  conf_links,
-)
-
 # 35-lang-normalize.conf
 orths = []
 foreach o : orth_files          # orth_files is from fc-lang/meson.build
