--- a/meson.build
+++ b/meson.build
@@ -25,16 +25,16 @@
 cc = meson.get_compiler('c')
 
 
-freetype_dep = dependency('freetype2', method: 'pkg-config', version: freetype_req, required: false)
+freetype_dep = dependency('freetype2', method: 'pkg-config', version: freetype_req, required: false, static: true)
 
 # Give another shot using CMake
 if not freetype_dep.found()
   freetype_dep = dependency('freetype', method: 'cmake', version: freetype_req_cmake,
-    fallback: ['freetype2', 'freetype_dep'])
+    fallback: ['freetype2', 'freetype_dep'], static: true)
 endif
 
 expat_dep = dependency('expat',
-  fallback: ['expat', 'expat_dep'])
+  fallback: ['expat', 'expat_dep'], static: true)
 
 i18n = import('i18n')
 pkgmod = import('pkgconfig')
@@ -216,7 +216,7 @@
     fonts_conf.set('FC_DEFAULT_FONTS', '<dir>/usr/share/fonts</dir>')
   endif
   fc_cachedir = join_paths(prefix, get_option('localstatedir'), 'cache', meson.project_name())
-  thread_dep = dependency('threads')
+  thread_dep = dependency('threads', static: true)
   conf.set('HAVE_PTHREAD', 1)
   deps += [thread_dep]
 endif
