diff --git a/lib/init/rc.boot b/lib/init/rc.boot
index f80a2f0..3923426 100755
--- a/lib/init/rc.boot
+++ b/lib/init/rc.boot
@@ -33,20 +33,8 @@ log "Loading rc.conf settings..."; {
 }
 
 log "Starting device manager..."; {
-    if command -v udevd >/dev/null; then
-        log "Starting udev..."
-
-        udevd -d
-        udevadm trigger -c add -t subsystems
-        udevadm trigger -c add -t devices
-        udevadm settle
-
-    elif command -v mdev >/dev/null; then
-        log "Starting mdev..."
-
-        mdev -s
-        mdev -df & mdev_pid=$!
-    fi
+    log "Starting mdev..."
+    mdev -s
 }
 
 log "Remounting rootfs as read-only..."; {
