diff --git a/lib/init/rc.boot b/lib/init/rc.boot
index f80a2f0..3923426 100755
--- a/lib/init/rc.boot
+++ b/lib/init/rc.boot
@@ -33,20 +33,8 @@ log "Loading rc.conf settings..."; {
 }
 
 log "Starting device manager..."; {
-    if command -v udevd >/dev/null; then
-        log "Starting udev..."
-
-        udevd -d
-        udevadm trigger -c add -t subsystems
-        udevadm trigger -c add -t devices
-        udevadm settle
-
-    elif command -v mdev >/dev/null; then
-        log "Starting mdev..."
-
-        mdev -s
-        mdev -df & mdev_pid=$!
-    fi
+    log "Starting mdev..."
+    mdev -s
 }
 
 log "Remounting rootfs as read-only..."; {
@@ -71,14 +59,6 @@ log "Mounting all local filesystems..."; {
     mount -a || sos
 }
 
-log "Enabling swap..."; {
-    swapon -a || sos
-}
-
-log "Seeding random..."; {
-    random_seed load
-}
-
 log "Setting up loopback..."; {
     ip link set up dev lo
 }
@@ -106,22 +86,6 @@ log "Loading sysctl settings..."; {
     done
 }
 
-log "Killing device manager to make way for service..."; {
-    if command -v udevd >/dev/null; then
-        udevadm control --exit
-
-    elif [ "$mdev_pid" ]; then
-        kill "$mdev_pid"
-
-        # Try to set the hotplug script to mdev.
-        # This will silently fail if unavailable.
-        #
-        # The user should then run the mdev service
-        # to enable hotplugging.
-        command -v mdev > /proc/sys/kernel/hotplug
-    fi 2>/dev/null
-}
-
 log "Running boot hooks..."; {
     run_hook boot
 }
