--- a/lib/init/rc.boot
+++ b/lib/init/rc.boot
@@ -4,7 +4,7 @@
 # Shared code between boot/shutdown.
 . /usr/lib/init/rc.lib
 
-log "Welcome to KISS!"
+log "Welcome Dilyn!"
 
 log "Mounting pseudo filesystems..."; {
     mnt nosuid,noexec,nodev    proc     proc /proc
@@ -33,20 +33,10 @@
 }
 
 log "Starting device manager..."; {
-    if command -v udevd >/dev/null; then
-        log "Starting udev..."
+    log "Starting mdev..."
 
-        udevd -d
-        udevadm trigger -c add -t subsystems
-        udevadm trigger -c add -t devices
-        udevadm settle
-
-    elif command -v mdev >/dev/null; then
-        log "Starting mdev..."
-
-        mdev -s
-        mdev -df & mdev_pid=$!
-    fi
+    mdev -s
+    mdev -df & mdev_pid=$!
 }
 
 log "Remounting rootfs as read-only..."; {
@@ -71,14 +61,6 @@
     mount -a || sos
 }
 
-log "Enabling swap..."; {
-    swapon -a || sos
-}
-
-log "Seeding random..."; {
-    random_seed load
-}
-
 log "Setting up loopback..."; {
     ip link set up dev lo
 }
@@ -106,22 +88,6 @@
     done
 }
 
-log "Killing device manager to make way for service..."; {
-    if command -v udevd >/dev/null; then
-        udevadm control --exit
-
-    elif [ "$mdev_pid" ]; then
-        kill "$mdev_pid"
-
-        # Try to set the hotplug script to mdev.
-        # This will silently fail if unavailable.
-        #
-        # The user should then run the mdev service
-        # to enable hotplugging.
-        command -v mdev > /proc/sys/kernel/hotplug
-    fi 2>/dev/null
-}
-
 log "Running boot hooks..."; {
     run_hook boot
 }
