--- a/dom/plugins/ipc/PluginInstanceParent.cpp
+++ b/dom/plugins/ipc/PluginInstanceParent.cpp
@@ -230,7 +230,7 @@
   XID id;
 #elif defined(XP_DARWIN)
   intptr_t id;
-#elif defined(ANDROID)
+#elif defined(ANDROID) || defined(MOZ_WAYLAND)
   // TODO: Need Android impl
   int id;
 #else
--- a/dom/plugins/ipc/PluginMessageUtils.h
+++ b/dom/plugins/ipc/PluginMessageUtils.h
@@ -115,7 +115,7 @@
 typedef HWND NativeWindowHandle;
 #elif defined(MOZ_X11)
 typedef XID NativeWindowHandle;
-#elif defined(XP_DARWIN) || defined(ANDROID)
+#elif defined(XP_DARWIN) || defined(ANDROID) || defined(MOZ_WAYLAND)
 typedef intptr_t NativeWindowHandle;  // never actually used, will always be 0
 #else
 #  error Need NativeWindowHandle for this platform
