--- a/Makefile
+++ b/Makefile
@@ -130,8 +130,8 @@
 
 WLROOTS_CFLAGS += -DWLR_USE_UNSTABLE=1
 
-PANGO_CFLAGS != pkg-config --cflags pangocairo
-PANGO_LIBS != pkg-config --libs pangocairo
+PANGO_CFLAGS != pkg-config --cflags --static pangocairo
+PANGO_LIBS != pkg-config --libs --static pangocairo
 
 CAIRO_CFLAGS != pkg-config --cflags cairo
 CAIRO_LIBS != pkg-config --libs cairo
@@ -142,24 +142,60 @@
 XKBCOMMON_CFLAGS != pkg-config --cflags xkbcommon
 XKBCOMMON_LIBS != pkg-config --libs xkbcommon
 
-WAYLAND_CFLAGS != pkg-config --cflags wayland-server
-WAYLAND_LIBS != pkg-config --libs wayland-server
+WAYLAND_S_CFLAGS != pkg-config --cflags wayland-server
+WAYLAND_S_LIBS != pkg-config --libs wayland-server
 
+WAYLAND_E_CFLAGS != pkg-config --cflags --static wayland-egl
+WAYLAND_E_LIBS != pkg-config --libs --static wayland-egl
+
 LIBINPUT_CFLAGS != pkg-config --cflags libinput
 LIBINPUT_LIBS != pkg-config --libs libinput
 
+UDEV_CFLAGS != pkg-config --cflags --static libudev
+UDEV_LIBS != pkg-config --libs --static libudev
+
+EVDEV_CFLAGS != pkg-config --cflags --static libevdev
+EVDEV_LIBS != pkg-config --libs --static libevdev
+
+MTDEV_CFLAGS != pkg-config --cflags --static mtdev
+MTDEV_LIBS != pkg-config --libs --static mtdev
+
 UCL_CFLAGS != pkg-config --cflags libucl
 UCL_LIBS != pkg-config --libs libucl
 
+EGL_CFLAGS != pkg-config --cflags --static egl
+EGL_LIBS != pkg-config --libs --static egl
+
+GLES_CFLAGS != pkg-config --cflags --static glesv2
+GLES_LIBS != pkg-config --libs --static glesv2
+
+DRM_CFLAGS != pkg-config --cflags --static libdrm
+DRM_LIBS != pkg-config --libs --static libdrm
+
+GBM_CFLAGS != pkg-config --cflags --static gbm
+GBM_LIBS != pkg-config --libs --static gbm
+
+ZLIB_CFLAGS != pkg-config --cflags --static zlib
+ZLIB_LIBS != pkg-config --libs --static zlib
+
 CFLAGS += \
 	${WLROOTS_CFLAGS} \
 	${PANGO_CFLAGS} \
 	${CAIRO_CFLAGS} \
 	${PIXMAN_CFLAGS} \
 	${XKBCOMMON_CFLAGS} \
-	${WAYLAND_CFLAGS} \
+	${WAYLAND_S_CFLAGS} \
+	${WAYLAND_E_CFLAGS} \
 	${LIBINPUT_CFLAGS} \
-	${UCL_CFLAGS}
+	${UDEV_CFLAGS} \
+	${EVDEV_CFLAGS} \
+	${MTDEV_CFLAGS} \
+	${UCL_CFLAGS} \
+	${EGL_CFLAGS} \
+	${GLES_CFLAGS} \
+	${DRM_CFLAGS} \
+	${GBM_CFLAGS} \
+	${ZLIB_CFLAGS}
 
 LIBS = \
 	${WLROOTS_LIBS} \
@@ -167,9 +203,18 @@
 	${CAIRO_LIBS} \
 	${PIXMAN_LIBS} \
 	${XKBCOMMON_LIBS} \
-	${WAYLAND_LIBS} \
+	${WAYLAND_S_LIBS} \
+	${WAYLAND_E_LIBS} \
 	${LIBINPUT_LIBS} \
-	${UCL_LIBS}
+	${UDEV_LIBS} \
+	${EVDEV_LIBS} \
+	${MTDEV_LIBS} \
+	${UCL_LIBS} \
+	${EGL_LIBS} \
+	${GLES_LIBS} \
+	${DRM_LIBS} \
+	${GBM_LIBS} \
+	${ZLIB_LIBS}
 
 PROTOCOL_HEADERS = xdg-shell-protocol.h
 
