--- a/meson.build
+++ b/meson.build
@@ -438,9 +438,9 @@
         configh_data.set10('HAVE_XKBCLI_INTERACTIVE_X11', true)
     endif
     if get_option('enable-wayland')
-        wayland_client_dep = dependency('wayland-client', version: '>=1.2.0', required: false)
-        wayland_protocols_dep = dependency('wayland-protocols', version: '>=1.12', required: false)
-        wayland_scanner_dep = dependency('wayland-scanner', required: false, native: true)
+        wayland_client_dep = dependency('wayland-client', version: '>=1.2.0', required: false, static: true)
+        wayland_protocols_dep = dependency('wayland-protocols', version: '>=1.12', required: false, static: true)
+        wayland_scanner_dep = dependency('wayland-scanner', required: false, native: true, static: true)
         if not wayland_client_dep.found() or not wayland_protocols_dep.found() or not wayland_scanner_dep.found()
             error('''The Wayland xkbcli programs require wayland-client >= 1.2.0, wayland-protocols >= 1.7 which were not found.
 You can disable the Wayland xkbcli programs with -Denable-wayland=false.''')
