--- a/kiss
+++ b/kiss
@@ -353,6 +353,10 @@
     if null "${2##git+*}"; then
         _res=$2
 
+    # Fossil repository.
+    elif null "${2##fossil+*}"; then
+        _res=$2
+
     # Remote source (cached).
     elif [ -f "$src_dir/$1/${3:+"$3/"}${2##*/}" ]; then
         _res=$src_dir/$1/${3:+"$3/"}${2##*/}
@@ -451,6 +455,15 @@
     git -c advice.detachedHead=0 checkout "${com:-FETCH_HEAD}"
 }
 
+pkg_source_fossil() {
+    log "$repo_name" "Cloning $1"
+
+    # Split the source into URL + OBJECT (branch or commit).
+    url=$1 com=${url##*[@#]} com=${com#${url%[#@]*}}
+
+    fossil open -f "${url%[#@]}" "${2:-trunk}"
+}
+
 pkg_source_tar() {
     # This is a portable shell implementation of GNU tar's
     # '--strip-components 1'. Use of this function denotes a
@@ -518,8 +531,9 @@
         null "$_res" || mkcd "$mak_dir/$1/$dest"
 
         case $_res in
-            git+*)
-                pkg_source_git "${_res##git+}"
+            git+*|fossil+*)
+                vcs="${src%%+*}"
+                "pkg_source_$vcs" "${_res##$vcs+}"
             ;;
 
             *.tar|*.tar.??|*.tar.???|*.tar.????|*.t?z)
@@ -1546,6 +1560,12 @@
             subm=$(git -C "$repo" rev-parse --show-superproject-working-tree)
             repo=$(git -C "${subm:-"$repo"}" rev-parse --show-toplevel)
 
+        elif fossil info "$repo" >/dev/null 2>&1; then
+            repo_type=fossil
+
+            # Get the Fossil repository root directory.
+            repo=$(fossil info | grep local-root | cut -d ':' -f 2)
+
         elif ! [ -d "$repo" ]; then
             continue
 
@@ -1577,8 +1597,11 @@
         # env:  PWD is path to repository
         run_hook pre-update "$#"
 
-        case $repo_type in git)
-            pkg_update_git "$@"
+        case $repo_type in
+            git)
+                pkg_update_git "$@" ;;
+            fossil)
+                pkg_update_fossil "$@" ;;
         esac
 
         # arg1: post-update
@@ -1597,6 +1620,10 @@
     "$@" git submodule update --remote --init -f
 }
 
+pkg_update_fossil() {
+    "$@" fossil pull && "$@" fossil update
+}
+
 pkg_upgrade() {
     log "Checking for new package versions"
     set +f
