--- a/kiss
+++ b/kiss
@@ -361,6 +361,12 @@
         _des=$src_dir/$1/${3:+"$3/"}${2##*/}
         _des=${_des%[@#]*}
 
+    # Fossil repository.
+    elif null "${2##fossil+*}"; then
+        _res=$2
+        _des=$src_dir/$1/${3:+"$3/"}${2##*/}
+        _des=${_des%[@#]*}
+
     # Remote source (cached).
     elif [ -f "$src_dir/$1/${3:+"$3/"}${2##*/}" ]; then
         _res=$src_dir/$1/${3:+"$3/"}${2##*/}
@@ -467,6 +473,24 @@
     git -c advice.detachedHead=0 checkout "${com:-FETCH_HEAD}"
 }
 
+pkg_source_fossil() {
+    com=${1##*[@#]}
+    com=${com#${1%[#@]*}}
+
+    log "$repo_name" "Checking out ${com:-trunk}"
+
+    [ -d .fslckout ] || {
+        fossil open -f "${1%[#@]*}" "${com:-trunk}"
+    }
+
+    # Only fetch latest changes if not <url>[#]<sha|tag>.
+    case $1 in *"#$com") ;; *)
+        fossil pull default "$com" || fossil pull
+    esac
+
+    fossil pull "${com:-trunk}"
+}
+
 pkg_source_tar() {
     # This is a portable shell implementation of GNU tar's
     # '--strip-components 1'. Use of this function denotes a
@@ -1644,8 +1668,11 @@
         # env:  PWD is path to repository
         run_hook pre-update "$#" "$user"
 
-        case $repo_type in git)
-            pkg_update_git "$@"
+        case $repo_type in
+            git)
+                pkg_update_git "$@" ;;
+            fossil)
+                pkg_update_fossil "$@" ;;
         esac
 
         # arg1: post-update
@@ -1664,6 +1691,10 @@
     "$@" git submodule update --remote --init -f
 }
 
+pkg_update_fossil() {
+    "$@" fossil pull && "$@" fossil update
+}
+
 pkg_upgrade() {
     log "Checking for new package versions"
     set +f
