--- a/contrib/kiss-outdated
+++ b/contrib/kiss-outdated
@@ -141,6 +141,10 @@
             remote=lux-brightness-ventto
         ;;
 
+        make)
+            remote=bmake
+            ;;
+
         man-pages)
             # TODO: Report to repology.org.
             # man-pages       -> manpages
--- a/contrib/kiss-orphans
+++ b/contrib/kiss-orphans
@@ -7,8 +7,8 @@
     case $pkg in
         # Exemptions for orphans which aren't really
         # orphans. Exclude them from the list.
-        baseinit|baselayout|gcc|e2fsprogs|musl|\
-        make|busybox|bzip2|grub|kiss|git)
+        b3sum|baseinit|baselayout|kiss|llvm|make|\
+        musl|nawk|oksh|openntpd|otools|sm|smdev|toybox|zfs|zstd)
             continue
     esac
 
