--- a/contrib/kiss-orphans
+++ b/contrib/kiss-orphans
@@ -7,8 +7,8 @@
     case $pkg in
         # Exemptions for orphans which aren't really
         # orphans. Exclude them from the list.
-        baseinit|baselayout|gcc|e2fsprogs|musl|\
-        make|busybox|bzip2|grub|kiss|git)
+        b3sum|baselayout|dash|hummingbird|kiss|llvm|\
+        make|musl|nawk|pax|openntpd|sm|toybox|zfs|zstd)
             continue
     esac
 
--- a/contrib/kiss-outdated
+++ b/contrib/kiss-outdated
@@ -75,6 +75,11 @@
             remote=boehm-gc
         ;;
 
+        gmake)
+            # shell-check is dumb here so this must be quoted.
+            remote='make'
+        ;;
+
         gnugrep)
             # shell-check is dumb here so this must be quoted.
             remote='grep'
@@ -133,6 +138,10 @@
             remote=links
         ;;
 
+        make)
+            remote=bmake
+        ;;
+
         linux-headers)
             remote=linux
         ;;
