--- a/Makefile
+++ b/Makefile
@@ -135,7 +135,6 @@
         $(filter-out -j, $(lastword -j1 $(filter -l% -j%, $(MAKEFLAGS)))) \
 
 ninja-cmd-goals = $(or $(MAKECMDGOALS), all)
-ninja-cmd-goals += $(foreach t, $(.tests), $(.test.deps.$t))
 
 makefile-targets := build.ninja ctags TAGS cscope dist clean uninstall
 # "ninja -t targets" also lists all prerequisites.  If build system
@@ -175,7 +174,6 @@
 
 SUBDIR_MAKEFLAGS=$(if $(V),,--no-print-directory --quiet)
 
-include $(SRC_PATH)/tests/Makefile.include
 
 all: recurse-all
 
@@ -273,15 +271,8 @@
 		-f"$(SRC_PATH)"/cscope.out, 		\
 		"cscope", "Re-index $(SRC_PATH)")
 
-# Needed by "meson install"
-export DESTDIR
 
-include $(SRC_PATH)/tests/docker/Makefile.include
-include $(SRC_PATH)/tests/vm/Makefile.include
 
-print-help-run = printf "  %-30s - %s\\n" "$1" "$2"
-print-help = @$(call print-help-run,$1,$2)
-
 .PHONY: help
 help:
 	@echo  'Generic targets:'
--- a/meson.build
+++ b/meson.build
@@ -2381,7 +2381,6 @@
 subdir('tools')
 subdir('pc-bios')
 subdir('docs')
-subdir('tests')
 if gtk.found()
   subdir('po')
 endif
