--- a/meson.build
+++ b/meson.build
@@ -75,7 +75,7 @@
 
 freetype_dep = null_dep
 if not get_option('freetype').disabled()
-  freetype_dep = dependency('freetype2', required: false)
+  freetype_dep = dependency('freetype2', required: false, static: true)
 
   if (not freetype_dep.found() and
       cpp.get_id() == 'msvc' and
@@ -96,7 +96,7 @@
 gobject_dep = dependency('gobject-2.0', required: get_option('gobject'),
                          fallback: ['glib', 'libgobject_dep'])
 fontconfig_dep = dependency('fontconfig', required: get_option('fontconfig'),
-                            fallback: ['fontconfig', 'fontconfig_dep'])
+                            fallback: ['fontconfig', 'fontconfig_dep'], static: true)
 graphite2_dep = dependency('graphite2', required: get_option('graphite'))
 
 icu_dep = null_dep
@@ -284,7 +284,7 @@
 # threads
 thread_dep = null_dep
 if host_machine.system() != 'windows'
-  thread_dep = dependency('threads', required: false)
+  thread_dep = dependency('threads', required: false, static: true)
 
   if thread_dep.found()
     conf.set('HAVE_PTHREAD', 1)
